src/GafferImage/ImageWriter.cpp

```diff
@@ -869,6 +935,12 @@ IECore::MurmurHash ImageWriter::hash( const Context *context ) const
 void ImageWriter::execute() const
 {
 	// Create an OIIO::ImageOutput
+	Context * c = const_cast<Context *>( Context::current() );
+	c->set( "gaffer:colorspace", defaultColorSpace()( currentFileFormat() ) );
+
+
+	Expression *e = const_cast<Expression *>( expression() );
```

Comment:
> Alright, sounds good!

Cool. Let's do that then.

> Could this ComputeNode even works without the context at all, based on input plug use the registered function to compute a colorspace name and set an outPlug with that value?

The problem would be that the ComputeNode would need access to all the plugs of the ImageWriter, but it only has access to its own plugs. In some ways it'd be nice if the ImageWriter was a ComputeNode as well as a TaskNode, but I don't think that's a good idea generally.

> Another thing I'm wondering is when you say remove the dependency on Python, you mean only the execution of the expression because correct me if I'm wrong but computing the colorSpace name still goes through python but I assume this is cached?

Correct. I'm suggesting we remove the dependency on Python at the _library_ level, which would mean it was still possible to use ImageWriter from a pure C++ application linked to libGafferImage. At the _application_ level, we still have the python dependency because the application is written in Python and the config uses python, but that's fine, that's how the main Gaffer app is written. We just want to keep the library clean so it could be used independently if necessary.

> Does the Expression using the Context also cache efficiently?

Yes. The hash for the python expression (which forms the key for the cache lookup) is always computed entirely in C++, this is what allows python expressions to outperform python nodes. In this case the hash will be composed only from the context variable, so we should execute the expression only once for each unique value of the variable.